### PR TITLE
Add TODO with 0.8 targets

### PR DESCRIPTION
**Pull request by jagerman**

### CODE REVIEW

File: TODO.md [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# TODO plan for 0.8
+
+These may not all be implemented, but are a tentatively roadmap for things to work for in 0.8.  0.8
```
Comment (notlesh) [inline TODO.md]:
```suggestion
These may not all be implemented, but are a tentative roadmap for things to work for in 0.8.  0.8
```

Comment (notlesh) [reply TODO.md]: :laughing:

File: TODO.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+investment paying off by cleaning up the code to allow fast and more reliable future features.
+
+## Installer build improvements
+
+Better installers so we can "make windows" or "make macos" and out pops an installer.  Currently
+installers are too manual and we need to automate this.  More specifically:
+- CPack installer for windows (32 and 64-bit).  If possible figure out how to do a "single"
+  installer that can install the right one.
+- macOS build automation (i.e. up to code signing)
+- lokinet GUI package for linux (debs)
```
Comment (notlesh) [inline TODO.md]: A fully-fledged `lokinet` installer should include the GUI, which is a separate project. I'm not sure how to best handle this...

File: TODO.md [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+## Clean up cruft.
+
+There is some unused (non-code) junk that has built up in the repo: (.exe files, random patches,
+etc.).
+
+
+## Make `make format` less annoying
+
+New `make format' rules.  Relax some things, decide on other sane rules that are helpful without
```
Comment (notlesh) [inline TODO.md]:
```suggestion
New `make format` rules.  Relax some things, decide on other sane rules that are helpful without
```
(use closing back tick)

File: TODO.md [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+annoying.
+
+## .ini file overhaul
+
+Right now we have a hard-coded generated .ini file that only includes some settings.  We need to
+overhaul this to include everything -- some options with initial settings, other options commented
+out -- along with descriptions of each option.  We should do this "right": i.e. by having one single
+set of specification of options that can do both parsing and config file generation rather than
+duplicating options in parsing and options printed in the generated config file (because that will
+inevitably diverge).
```
Comment (notlesh) [inline TODO.md]: If we're doing anything more than just generating commented-out lines for the default/example config, I'd like to consider removing some of the side effects in the config parsing code (notably, parsing the conf will change the log level).

File: TODO.md [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+## Refactoring:
+- DHT code.  Two phases:
+  - phase 1, while documenting, make notes about refactoring DHT
+  - phase 2: implement DHT refactor.
+- LokiMQ integration.
+  - replace libabyss RPC interface
+  - lokid communication
+  - (experimental) handle job scheduling
+  - (experimental) work incoming UDP polling into lokimq's poll loop (potentially replacing libuv)
+
```
Comment (notlesh) [inline TODO.md]: Note: replacing the RPC interface will require updates to the GUI and `munin`

Comment (jagerman) [reply TODO.md]: Right, but they shouldn't be *too* bad: it can still be JSON on the wire, just not over HTTP anymore.

**Review by melyalex — APPROVED** [review]

**Review by melyalex — APPROVED** [review]

**Pull request closed without merging**